stan/math/rev/mat/functor/cvodes_ode_data.hpp

```diff
@@ -165,14 +165,13 @@ class cvodes_ode_data {
    * y to be the initial of the coupled ode system.
    */
   inline int jacobian_states(double t, const double y[], SUNMatrix J) const {
-    const std::vector<double> y_vec(y, y + N_);
     start_nested();
-    std::vector<var> y_vec_var(y_vec.begin(), y_vec.end());
+    const std::vector<var> y_vec_var(y, y + N_);
     coupled_ode_system<F, var, double> ode_jacobian(f_, y_vec_var, theta_dbl_,
                                                     x_, x_int_, msgs_);
-    std::vector<double> jacobian_y(ode_jacobian.size(), 0);
+    std::vector<double>&& jacobian_y = std::vector<double>(ode_jacobian.size());
```

Comment:
Could you explain why you're using R-value references here? I'm not very familiar with `&&`, but my understanding is that you are creating a new object on the R.H.S of the `=` operator anyway. In that case, the old and the new statement would be doing the same thing, i.e. compiled into the same code.

Reply:
This is again an attempt to let the compiler optimise better. We do not need the Jacobian stored in this function at all. Telling the compiler that the vector `Jacobean_y` can be treated as r-value should allow the compiler to figure out that we actually want things to be written into `J`, because that where the r-value ends up. Thus, this avoids copying.